functional_widget/lib/function_to_widget_class.dart

```diff
@@ -250,23 +251,27 @@ class FunctionalWidgetGenerator
     List<TypeParameterElement> typeParameters,
   ) {
     return typeParameters.map((e) {
-      final displayName = e.bound
-          ?.getDisplayString(withNullability: false /* TODO upgrade to true */);
+      final displayName = e.bound?.getDisplayString(withNullability: true);
       return displayName != null
           ? refer('${e.displayName} extends $displayName')
           : refer(e.displayName);
     });
   }
 
-  Constructor _getConstructor(List<Parameter> fields, {String doc}) {
+  Constructor _getConstructor(
+    List<Parameter> fields, {
+    String? doc,
+    bool keyIsRequired = false,
+  }) {
     return Constructor(
       (b) => b
         ..constant = true
         ..optionalParameters.add(Parameter((b) => b
+          ..required = keyIsRequired
           ..named = true
           ..name = 'key'
           ..docs.clear()
-          ..type = _keyRef))
+          ..type = keyIsRequired ? _keyRef : _optionalKeyRef))
```

Comment:
Better way of doing this?

Reply:
I think a `TypeReference` builder would be the right thing to use here.

Can set `isNullable` to `keyIsRequired`.